ipam.py

```diff
@@ -95,15 +119,25 @@ def _assign_address(self, handle_id):
                            msg="Failed to assign IP address",
                            details=e.message)
         else:
-            try:
-                ipv4 = ipv4_addrs[0]
-            except IndexError:
-                _log.error("No IPv4 address returned, exiting")
-                raise CniError(ERR_CODE_GENERIC,
-                               msg="No IPv4 addresses available in pool")
+            if num_v4:
+                try:
+                    ipv4 = ipv4_addrs[0]
```

Comment:
What about `ipv4 = IPNetwork(ipv4_addrs[0])` then you can simplify your return statement below.